File: backend/routes/copilot.py
from fastapi import APIRouter
from pydantic import BaseModel

from services.copilot import ask_copilot

router = APIRouter()


class ChatRequest(BaseModel):
    message: str


@router.post("/copilot")
def chat(request: ChatRequest):

    answer = ask_copilot(request.message)

    return {
        "answer": answer
    }

File: frontend/lib/copilot.ts
import api from "./api";

export async function askCopilot(
  message: string
): Promise<string> {
  try {
    const response = await api.post("/copilot", {
      message: message,
    });

    return response.data.answer;

  } catch (error: any) {
    console.error("Copilot API Error:", error);

    if (error.response) {
      console.error(
        "Backend response:",
        error.response.data
      );
    }

    throw error;
  }
}